inginious/frontend/plugins/task_hints/pages/api/edit_hints_api.py

```diff
@@ -0,0 +1,40 @@
+import json
+import os
+import web
+
+from inginious.frontend.pages.api._api_page import APIAuthenticatedPage, APIError
+from inginious.frontend.plugins.utils import get_mandatory_parameter
+from inginious.common.course_factory import InvalidNameException, CourseNotFoundException
+
+class EditHintsAPI(APIAuthenticatedPage):
+
+    def API_GET(self):
+
+        """
+            Get all the data for saved hints in task
+        """
+
+        input_data = web.input()
+        
+        course_id = get_mandatory_parameter(input_data, 'course_id')
+        task_id = get_mandatory_parameter(input_data, 'task_id')
+        username = self.user_manager.session_username()
+
+        try:
+            course = self.course_factory.get_course(course_id)
+        except (InvalidNameException, CourseNotFoundException):
+            raise APIError(400, {"error": "The course does not exists or the user does not have permissions"})
+
+        try: 
+            task = self.task_factory.get_task(course, task_id)
+        except Exception:
+            raise APIError(400, {"error": "The task does not exist in the course"})
+
+
+        if not self.user_manager.course_is_user_registered(course, username):
+            raise APIError(400, {"error": "The course does not exists or the user does not have permissions"})
```

Comment:
I did it to check that the user has permissions to edit and save the task, however, I removed this check since this is done with the course check when the task is saved.